recipes/make/all/conanfile.py

```diff
@@ -62,3 +62,4 @@ def package_info(self):
 
         self.output.info('Creating CONAN_MAKE_PROGRAM environment variable: %s' % make)
         self.env_info.CONAN_MAKE_PROGRAM = make
+        self.env_info.CMAKE_MAKE_PROGRAM = make
```

Comment:
AFAIK, CMake doesn't listen this env variable, it's an argument for CMake configuration, it's different.

But you can set `tools.gnu:make_program` for `CMakeToolchain`.

Reply:
Compiling zlib with tools comming from conan gives me this error:

```
zlib/1.2.11: CMake command: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/root/.conan/data/zlib/1.2.11/_/_/build/dfbe50feef7f3c6223a476cd5aeadb687084a646/build/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/root/.conan/data/zlib/1.2.11/_/_/package/dfbe50feef7f3c6223a476cd5aeadb687084a646" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/root/.conan/data/zlib/1.2.11/_/_/build/dfbe50feef7f3c6223a476cd5aeadb687084a646/src"
-- Using Conan toolchain: /root/.conan/data/zlib/1.2.11/_/_/build/dfbe50feef7f3c6223a476cd5aeadb687084a646/build/generators/conan_toolchain.cmake
-- Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
```

Reply:
Please, checkout my Dockerfile here
https://github.com/AndreyMlashkin/minimalistic_conan_docker

Reply:
> Compiling zlib with tools comming from conan gives me this error:
> 
> ```
> zlib/1.2.11: CMake command: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/root/.conan/data/zlib/1.2.11/_/_/build/dfbe50feef7f3c6223a476cd5aeadb687084a646/build/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/root/.conan/data/zlib/1.2.11/_/_/package/dfbe50feef7f3c6223a476cd5aeadb687084a646" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/root/.conan/data/zlib/1.2.11/_/_/build/dfbe50feef7f3c6223a476cd5aeadb687084a646/src"
> -- Using Conan toolchain: /root/.conan/data/zlib/1.2.11/_/_/build/dfbe50feef7f3c6223a476cd5aeadb687084a646/build/generators/conan_toolchain.cmake
> -- Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)
> -- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
> CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
> CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
> ```

> But you can set `tools.gnu:make_program` conf for CMake & Autotools conan v2 build helpers.

Reply:
Check it again. I had Ninja there instead of make. Now it should demonstrate, that make does not work without setting a CMAKE_MAKE_PROGRAM variable

Reply:
Did you try to update `make` recipe with suggestion above?

Reply:
Hmmm... It didn't help, indeed.

Reply:
Are conf settings passed with the old generators? 🤔  I'd expect that to be a toolchain thing

Reply:
What should I use instead of AutoToolsBuildEnvironment?